### PR TITLE
Add runtime control over OptionROM loading

### PR DESCRIPTION
**Pull request by SergiiDmytruk**

### CODE REVIEW

File: UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c [hunk 1]
```diff
@@ -43,6 +45,88 @@ PciPlatformPrepController(
...
+  if (!EFI_ERROR (Status)) {
+      LoadOptionRom = IS_PCI_DISPLAY (&PciConfHeader);
+  }
+
+  gBS->CloseProtocol (
+      PciHandle,
+      &gEfiPciIoProtocolGuid,
+      PciIo,
+      PciHandle
+      );
```
Comment (miczyg1) [inline UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c]: We are not doing OpenProtocol, just HandleProtocol, so we shouldn't close the protocol AFAIK @SergiiDmytruk

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c (hunk 1 above)
Comment (SergiiDmytruk) [reply UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c]: Looks like you're right.  I wasn't sure about it, `PciGetPciRom()` does call `CloseProtocol()` after `HandleProtocol()` and I forgot to check with the spec.

Comment (miczyg1) [reply UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c]: I have double checked the spec and it lookes like OpenProtocol is an extended version of HandleProtocol. The difference is that HandleProtocol does not record anywhere if some agent is consuming a protocol. OpenProtocol does record which agent is cnsuming the protocol in a handle database and in such case it CloseProtocol should be required. So we can either:

1. Drop the CloseProtocol, OR
2. Replace HandleProtocol calls with OpenProtocol with attribute EFI_OPEN_PROTOCOL_BY_HANDLE_PROTOCOL

Comment (miczyg1) [reply UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c]: OK, I see you went with option 1. Closing discussion

**Review by miczyg1 — APPROVED** [review]

**Comment by miczyg1** [comment]
Two more suggestions to address here: https://github.com/Dasharo/DasharoModulePkg/pull/26
Before this PR can be merged

**Pull request merged**